### PR TITLE
Controller

### PR DESCRIPTION
**Pull request by MariusTarnaru**
modified CustomerController, employee.fxml  and create EmployeeController to read all Product fields from database and write in TableView from all users.

### CODE REVIEW

**Review by sabiutayo — DISMISSED** [review]

File: src/main/java/sample/Controller.java [hunk 1]
```diff
@@ -25,20 +26,21 @@
...
-        if (userVerifier.userExist(user)) {
+        if (userVerifier.userExist(user) && userVerifier.isUserEmployeeType(user)) {
             Parent parent = FXMLLoader.load(getClass().getResource("/employee.fxml"));
             Scene scene = new Scene(parent);
             Stage appStage = (Stage) ((Node) event.getSource()).getScene().getWindow();
             appStage.setScene(scene);
             appStage.show();
 
-        } else if (userVerifier.userExist(user) || userVerifier.isUserEmployeeType(user)) {
+        } else if (userVerifier.userExist(user) && !userVerifier.isUserEmployeeType(user)) {
```
Comment (stefan2212) [inline src/main/java/sample/Controller.java]: try to use the database interogation only once.  When you are calling userVerifier.userExists you also use the database

**Review by stefan2212 — CHANGES_REQUESTED** [review]

File: src/main/java/sample/CustomerController.java [hunk 2]
```diff
@@ -30,9 +32,16 @@ void onReturnClicked() throws IOException {
         Main.showParentScene();
     }
 
-    public static List<Product> nameOfProducts(Product product) {
+    @FXML
+    public void initialize() {
+        nameOfProduct.setCellValueFactory(new PropertyValueFactory<>("productName"));
+        priceOfProduct.setCellValueFactory(new PropertyValueFactory<>("price"));
+        quantity.setCellValueFactory(new PropertyValueFactory<>("quantity"));
```
Comment (sabiutayo) [inline src/main/java/sample/CustomerController.java]: de ce nu folosesti metodele din productDao  si UserDao

**Review by Bossy12 — APPROVED** [review]

**Comment by MariusTarnaru** [comment]
- I use ProductDao from TableView;
- i changed caller userVerifier.userExists to query database only once :)

**Review by Bossy12 — APPROVED** [review]

**Review by sabiutayo — APPROVED** [review]

File: src/main/java/sample/CustomerController.java [hunk 3]
```diff
@@ -30,9 +33,15 @@ void onReturnClicked() throws IOException {
...
-        Query<Product> query = session.createQuery(
-                "select p from Product p", Product.class);
-        return query.getResultList();
+    @FXML
+    public void initialize() {
+        nameOfProduct.setCellValueFactory(new PropertyValueFactory<>("productName"));
+        priceOfProduct.setCellValueFactory(new PropertyValueFactory<>("price"));
+        quantity.setCellValueFactory(new PropertyValueFactory<>("quantity"));
+        ProductDao products = new ProductDao();
+        List<ProductDao> result = products.getAllProducts();
```
Comment (stefan2212) [inline src/main/java/sample/CustomerController.java:68]: The logic where you bring the products should be in a separate method

File: src/main/java/sample/EmployeeController.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @FXML
+    public void initialize() {
+        nameOfProduct.setCellValueFactory(new PropertyValueFactory<>("productName"));
+        priceOfProduct.setCellValueFactory(new PropertyValueFactory<>("price"));
+        quantity.setCellValueFactory(new PropertyValueFactory<>("quantity"));
+
+        ProductDao products = new ProductDao();
+        List<ProductDao> result = products.getAllProducts();
```
Comment (stefan2212) [inline src/main/java/sample/EmployeeController.java:58]: Same here

**Review by stefan2212 — CHANGES_REQUESTED** [review]

**Review by Bossy12 — APPROVED** [review]

**Review by stefan2212 — APPROVED** [review]

**Pull request merged**